tests/common/unittest_common.cc

```diff
@@ -880,6 +888,135 @@ TEST (commonTensorsInfo, getNameInvalidParam1_n)
   EXPECT_EQ (0U, gst_tensors_info_get_names_string (&info));
 }
 
+/**
+ * @brief Test for printing tensors info with invalid param.
+ */
+TEST (commonTensorsInfo, printInvalidParam_n)
+{
+  EXPECT_EQ (NULL, gst_tensors_info_to_string (NULL));
+}
+
+/**
+ * @brief Test for printing tensors info with invalid index.
+ */
+TEST (commonTensorsInfo, printInvalidIndex_n)
+{
+  GstTensorsInfo info;
+  gchar *str, *pos;
+
+  gst_tensors_info_init (&info);
+  info.num_tensors = NNS_TENSOR_SIZE_LIMIT + 1;
+
+  str = gst_tensors_info_to_string (&info);
+
+  pos = g_strstr_len (str, -1, "Num_Tensors out of bound");
+  EXPECT_TRUE (pos != NULL);
+
+  g_free (str);
+}
+
+/**
+ * @brief Test for printing tensors info comparison.
+ */
+TEST (commonTensorsInfo, compareString)
+{
+  GstTensorsInfo info1, info2;
+  GstTensorInfo *_info1, *_info2;
+  gchar *str, *pos;
+
+  fill_tensors_info_for_test (&info1, &info2);
+
+  /* same info */
+  str = gst_tensors_info_compare_to_string (&info1, &info2);
+
+  pos = g_strstr_len (str, -1, "Not equal");
+  EXPECT_TRUE (pos == NULL);
+
+  g_free (str);
+
+  /* change number of tensors */
+  _info1 = gst_tensors_info_get_nth_info (&info1, info1.num_tensors);
+  _info1->type = _NNS_INT32;
+  _info1->dimension[0] = 1;
+  info1.num_tensors++;
+
+  str = gst_tensors_info_compare_to_string (&info1, &info2);
+
+  pos = g_strstr_len (str, -1, "Not equal");
+  EXPECT_TRUE (pos != NULL);
+  pos = g_strstr_len (pos + 9, -1, "Not equal");
+  EXPECT_TRUE (pos == NULL);
+
+  info1.num_tensors--;
+  g_free (str);
+
+  _info1 = gst_tensors_info_get_nth_info (&info2, 0);
+  _info2 = gst_tensors_info_get_nth_info (&info2, 1);
+
+  /* change tensor name */
+  _info1->name = g_strdup ("test-tensor1");
+  _info2->name = g_strdup ("test-tensor2");
+
+  str = gst_tensors_info_compare_to_string (&info1, &info2);
+
+  pos = g_strstr_len (str, -1, "Not equal");
+  EXPECT_TRUE (pos != NULL);
+  pos = g_strstr_len (pos + 9, -1, "Not equal");
+  EXPECT_TRUE (pos != NULL);
+  pos = g_strstr_len (pos + 9, -1, "Not equal");
+  EXPECT_TRUE (pos == NULL);
```

Comment:
How about this?
```suggestion
  gchar **results;
  guint err_cnt;
  
  results = g_strsplit (results, "Not equal", -1);
  err_cnt = g_strv_length (results);
  EXPECT_EQ (3U, err_cnt);
  
  g_strfreev (results );
```

Reply:
It is better adding new internal function to find key string :)